packages/utils/telemetry-utils/src/batchedTelemetryHelper.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+type ICustomDataMap = Record<string, number>;
```

Comment:
nitpick: This class has no "non" custom data. So, a better name for this should be `ITelemetryMetric` `ITelemetryMetricMap` but the word map is implied in the type which makes it kinda redundant.

Reply:
As Yann mentioned in a comment below here, there one issue with the current design of this type.
Let's imagine that you have metrics A, B and C and set your threshold to 5. Let's say you want to have metrics batched for 50 samples of each of those three metrics. 

What if the way your customers code is designed, you have to call `log()` with one of the metrics at a time?

Your customer might call log() in the following way `log({'A': 3}); log({'A': 7}); log({'B': 3}); log({'C': 9}); log({'A': 1});` 
But now, your batched metrics have 3 samples of A, 1 sample of B and 1 sample of C, your counter will hit 5 and you will send the data. 

So, you might want to adjust the logic here so you are sending the data when you hit X samples of customer defined metrics rather than X number of calls of the `log()` function

Reply:
Having a contract where it's okay to call `log` without all metrics are the same time can be achieved by creating several loggers instead of one. At the end of the day, we should be logging information in coherent groups. If A and B are independent enough that we want to call `log` for them separately, then it's probably preferable to have them be isolated in separate loggers.

I'm kind of thinking at a high level here, so I might be failing to consider some compelling scenario. If such a scenario arises, I would want to reconsider my position and perhaps advocate for something different. Please let me know if you can think of a scenario.

Reply:
@yann-achard-MS 

Sean and I had a discussion about this earlier today, and concluded that we can go ahead with your original proposal. Which is to design a logger with a single global counter but instead enforce stricter typing to the `log()` (so that it accepts the entire set for every invocation).

And as you said, I think we could we could create multiple loggers for metrics that are separately retrievable.